### PR TITLE
Misc commit status API fixes

### LINKED ISSUE #1128: Various minor problems with commit status implementation
There are various minor problems in the commit status API code, mostly left over from the original implementation prior to the overhaul.

### PR DESCRIPTION
**Pull request by cvrebert**
Fixes #1128 by addressing these various minor issues.

### CODE REVIEW

**Comment by dmarkov** [comment]
@cvrebert Thanks, someone will review your pull request soon

**Comment by dmarkov** [comment]
@pinaf please review

File: src/main/java/com/jcabi/github/mock/MkStatuses.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import javax.json.JsonObject;
+import javax.validation.constraints.NotNull;
+import lombok.EqualsAndHashCode;
+
+/**
+ * Mock of GitHub commit statuses.
+ * @author Chris Rebert (github@rebertia.com)
+ * @version $Id$
+ * @since 0.24
+ * @todo Finish implementing this class (MkStatuses), a mock of GitHub's
```
Comment (pinaf) [inline src/main/java/com/jcabi/github/mock/MkStatuses.java]: @cvrebert we need the issue number and time estimate here. `@todo #1129:30min blah...`

File: src/main/java/com/jcabi/github/mock/MkCommits.java [hunk 2]
```diff
@@ -115,9 +115,10 @@ public Commit get(
     }
 
     @Override
+    @NotNull(message = "statuses is never NULL")
     public Statuses statuses(
             @NotNull(message = "sha can't be NULL") final String sha
```
Comment (pinaf) [inline src/main/java/com/jcabi/github/mock/MkCommits.java]: @cvrebert let's bring this indentation back a bit

**Comment by pinaf** [comment]
@cvrebert 2 comments above

**Comment by pinaf** [comment]
@cvrebert why was `Statuses.Smart` removed?

**Comment by cvrebert** [comment]
> why was Statuses.Smart removed?

@pinaf Because after removing 2 broken methods from it, all the remaining methods just delegated to the Statuses object. It offers no additional methods/features compared to a plain Statuses object, so there's no reason for it to exist.

File: src/main/java/com/jcabi/github/mock/MkCommits.java (hunk 2 above)
Comment (cvrebert) [inline src/main/java/com/jcabi/github/mock/MkCommits.java]: Done

File: src/main/java/com/jcabi/github/mock/MkStatuses.java (hunk 1 above)
Comment (cvrebert) [inline src/main/java/com/jcabi/github/mock/MkStatuses.java]: Done

**Comment by cvrebert** [comment]
@pinaf I've responded to all of your comments.

**Comment by pinaf** [comment]
@cvrebert thanks

**Comment by pinaf** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by dmarkov** [comment]
@cvrebert there is a puzzle in this code `1129-3099cf0b`/#1133, we'll resolve it later

**Comment by dmarkov** [comment]
@pinaf Thanks for your contribution, **19 mins** was added to your account, payment ID is `AP-0RN2846820363222D`, 24 hours and 42 mins spent total. review comments (c=4) added as a bonus. +19 added to your rating, current score is: [+7295](http://www.netbout.com/b/35128?open=rating)

**Comment by dmarkov** [comment]
@rultor deploy now pls

**Comment by 0pdd** [comment]
@cvrebert the puzzle [#unknown]() is still not solved.

**Comment by 0pdd** [comment]
@cvrebert the puzzle [#1785](https://github.com/jcabi/jcabi-github/issues/1785) is still not solved.

**Pull request merged**